samples/net/sockets/civetweb_websocket_server/src/http_server_handlers.c

```diff
@@ -0,0 +1,232 @@
+		.cbdata = NULL
+	},
+	{
+		.url = URL_INDEX_HTML,
+		.hnd = this_index_html_handler,
+		.cbdata = NULL
+	},
+	{
+		.url = URL_INDEX_CSS,
+		.hnd = this_index_css_handler,
+		.cbdata = NULL
+	},
+	{
+		.url = URL_WS_JS,
+		.hnd = this_ws_js_handler,
+		.cbdata = NULL
+	},
+	{
+		.url = URL_FAVICON_ICO,
+		.hnd = this_favicon_ico_handler,
+		.cbdata = NULL
+	}
+};
+
+void init_http_server_handlers(struct mg_context *ctx)
+{
+	this_register_handlers(ctx, this_hndlrs, ARRAY_SIZE(this_hndlrs));
+}
+
+__code_decl void this_register_handlers(struct mg_context *ctx,
+					const http_handler_t *hndlrs_array,
+					const size_t hndlrs_array_len)
+{
+	for (size_t i = 0; i < hndlrs_array_len; i++) {
+		mg_set_request_handler(ctx,
+					hndlrs_array[i].url,
+					hndlrs_array[i].hnd,
+					hndlrs_array[i].cbdata);
+	}
+}
+
+__code_decl int this_send_buffer_chunked(struct mg_connection *conn,
+					 const char *mime_type,
+					 const char *buff,
+					 const size_t buff_len)
+{
+	/* TODO: Add ret after each function call! */
+	int ret = 0;
+
+	ret = mg_send_http_ok(conn, mime_type, -1);
+	if (ret < 0) {
+		goto error_this_send_buffer_chunked;
+	}
+
+	long long left_bytes = buff_len;
+	char *itr = (char *)buff;  /* buffer iterator */
+
+	LOG_DBG("Transferring:");
+	LOG_DBG("itr: 0x%08X ret: %d left_bytes: %lld chunk_size: %zd B",
+		(unsigned int)itr, ret, left_bytes, TX_CHUNK_SIZE_BYTES);
+
+	while (left_bytes > TX_CHUNK_SIZE_BYTES) {
+		ret = mg_send_chunk(conn, itr, TX_CHUNK_SIZE_BYTES);
+		itr += TX_CHUNK_SIZE_BYTES;
+		left_bytes -= TX_CHUNK_SIZE_BYTES;
+
+		LOG_DBG("itr: 0x%08X ret: %d left_bytes: %lld",
+			(unsigned int)itr, ret, left_bytes);
+
+		if (ret < 0) {
+			goto error_this_send_buffer_chunked;
+		}
+	}
+
+	if (left_bytes > 0) {
+		ret = mg_send_chunk(conn, itr, left_bytes);
+		itr += left_bytes;
+		left_bytes = 0;
+
+		LOG_DBG("itr: 0x%08X ret: %d left_bytes: %lld",
+			(unsigned int)itr, ret, left_bytes);
+
+		if (ret < 0) {
+			goto error_this_send_buffer_chunked;
+		}
+	}
+
+	/* Must be sent at the end of the chuked sequence */
+	ret = mg_send_chunk(conn, "", 0);
+
+error_this_send_buffer_chunked:
+	if (ret < 0) {
+		LOG_ERR("aborted! ret: %d", ret);
+	}
+
+	return ret;
+}
+
+__code_decl int this_redirect_2_index_html(struct mg_connection *conn,
+					   void *cbdata)
+{
+	int ret = 0;
+
+	ret = mg_send_http_redirect(conn, URL_INDEX_HTML, 303);
+	return 202;
+}
+
+__code_decl int this_index_html_handler(struct mg_connection *conn,
+					void *cbdata)
+{
+	__data_decl const char index_html[] = {
+#include "web_page/index.html.gz.inc"
+	};
+
+	int ret = 0;
+
+	ret = this_send_buffer_chunked(conn, HTTP_TEXT_HTML
+					     HTTP_EOFL
+					     HTTP_CONTEND_ENCODING
+					     HTTP_ENCODING_GZ,
+					     index_html, sizeof(index_html));
+	this_set_return_value(&ret);
+	return ret;
+}
+
+__code_decl int this_index_css_handler(struct mg_connection *conn, void *cbdata)
+{
+	__data_decl const char index_css[] = {
```

Comment:
same here

Reply:
The define of `__data_decl` is changed to `static` now. Thanks for advice!